File: 0766-toeplitz-matrix/0766-toeplitz-matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        m, n = len(matrix), len(matrix[0])
        for i in range(m):
            val = matrix[i][0]
            for k in range(1, min(m, n)):
                if (i+k) >= m:
                    break
                if matrix[i+k][k] != val:
                    return False
        for j in range(1, n):
            val = matrix[0][j]
            for k in range(1, min(m, n)):
                if (j+k) >= n:
                    break
                if matrix[k][j+k] != val:
                    return False
        return True

File: 0766-toeplitz-matrix/0766-toeplitz-matrix.rs
use std::cmp;
impl Solution {
    pub fn is_toeplitz_matrix(matrix: Vec<Vec<i32>>) -> bool {
        let m: i32 = matrix.len() as i32;
        let n: i32 = matrix[0].len() as i32;
        for i in 0..m as usize {
            let val: i32 = matrix[i][0];
            for k in 1..cmp::min(m, n) as usize {
                if (((i+k) as i32) >= m) {
                    break;
                }
                if (matrix[i+k][k] != val) {
                    return false;
                }
            }
        }
        for j in 1..n as usize {
            let val: i32 = matrix[0][j];
            for k in 1..cmp::min(m, n) as usize {
                if (((j+k) as i32) >= n) {
                    break;
                }
                if (matrix[k][j+k] != val) {
                    return false;
                }
            }
        }
        return true;
    }
}